### PR TITLE
3_envvars_insertion

### PR DESCRIPTION
**Pull request by ioannnikulin**

### CODE REVIEW

**Comment by cr3mebrulee** [comment]
Tests:
1. list is empty
2. less than first node: at the beginning
3. more then last node: at the end
4. between:
4.1 same value
4.2 before
	4.2.1 before the last node
	4.2.2. between the last and first node
4.3 empty node

File: libft/sources/map/mapss/ft_mapss_insert.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/08/09 13:20:33 by taretiuk          #+#    #+#             */
+/*   Updated: 2024/08/14 21:06:42 by inikulin         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../../libft.h"
+#include <stdlib.h>
+
+int	ft_mapss_insert(t_mapss *map, t_dlist *node)
```
Comment (ioannnikulin) [inline libft/sources/map/mapss/ft_mapss_insert.c:16]: TODO: please split into separate functions inside this very file - to meet the Norm

File: libft/tests/map/mapss/ft_mapss_add_test.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	char	*keys_expected[20];
+	char	*values_expected[20];
+}	t_testcase;
+
+static void	generate_tests(t_testcase *tests)
+{/*
+	const char *keys[] = {"PATH", "SHELL", "HOME", "USERNAME", "PATH", "SHLVL", "OLDPWD", "DISPLAY", ""};
+	const char *values[] = {"/usr/local/sbin", "/bin/bash", "/home/user", "/usr/local/sbin", "1", "/home/taretiuk/42/student", ":0", "X Window", "Error"};
+*/
+	tests[0] = (t_testcase){"PATH", "/usr/local/sbin", 1, {"PATH"}, {"/usr/local/sbin"}};
```
Comment (ioannnikulin) [inline libft/tests/map/mapss/ft_mapss_add_test.c:32]: TODO: please expand

**Comment by ioannnikulin** [comment]
There is A LOT of small refactoring in libft. Mostly splitting big files that call test functions into smaller files for readability. Please stay calm.

Main points of real interest right here are:
1. libft/**sources**/map/mapss/ft_mapss_insert.c
2. libft/**tests**/map/mapss/ft_mapss_insert_test.c
3. also you can now start those tests. Navigate into libft folder, `make all goodtests`. And then `./tests/test`, or `gdbtui ./tests/test`, then `b ft_mapss_add_test` [Enter], `start`, `n` or `s` many times

**Pull request merged**